pkg/virt-controller/watch/vmi.go

```diff
@@ -390,6 +390,41 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 
 		patchOps := []string{}
 
+		vmiNetworksByName := map[string]virtv1.Network{}
+		for _, network := range vmiCopy.Spec.Networks {
+			if network.Name != "" {
+				vmiNetworksByName[network.Name] = network
+			}
+		}
+		for i, iface := range vmiCopy.Status.Interfaces {
+			if vmiNetworksByName[iface.Name].NetworkSource.Pod != nil && vmiCopy.Status.Interfaces[i].IP != pod.Status.PodIP {
+				vmiCopy.Status.Interfaces[i].IP = pod.Status.PodIP
+
+				// For forwarding binding mechanism like masquerade,
+				// it's sufficient to expose only PodIP to list of IPs
+				vmiCopy.Status.Interfaces[i].IPs = []string{pod.Status.PodIP}
```

Comment:
don't we still have two components (virt-handler and virt-controller) simultaneously attempting to set this same value? Doesn't this conflict?

Reply:
@davidvossel I don't think so
1. I see that virt-handler only reuses the interfaces if they're present in vmi status https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L451
2. If Migrating VMI has Guest-agent, that is already handled and won't be updating from virt-handler  https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L470
3. Migrations are allowed with only masquerede so I don't see any other conflicts happening in theory https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L1619

Still, do you see any other places that can cause conflict?

Reply:
> Still, do you see any other places that can cause conflict?

it's difficult to tell, and the risk is high here. 

The `vmi.Status.Interfaces` array is written to by virt-handler when the VMI is in the Running phase, which means that virt-handler effectively owns those fields.

If there's ever an instance where somehow virt-handler and virt-controller are not in alignment on what value should be set on this field, we'll get into a reconcile loop storm where virt-handler and virt-controller continually attempt to override each others values.

Basically, this logic is complex enough that even if it's right today, and you've gotten every if statement separated between what values virt-handler and virt-controller sets, it's still fragile logic. We run the risk of breaking this or making a subtle change to the logic in the future which could impact all of KubeVirt. 

Is it possible to get the same results by setting the interface IP from virt-handler instead of virt-controller?

Reply:
> > Still, do you see any other places that can cause conflict?
> 
> it's difficult to tell, and the risk is high here.
> 
> The `vmi.Status.Interfaces` array is written to by virt-handler when the VMI is in the Running phase, which means that virt-handler effectively owns those fields.
> 
> If there's ever an instance where somehow virt-handler and virt-controller are not in alignment on what value should be set on this field, we'll get into a reconcile loop storm where virt-handler and virt-controller continually attempt to override each others values.
> 
> Basically, this logic is complex enough that even if it's right today, and you've gotten every if statement separated between what values virt-handler and virt-controller sets, it's still fragile logic. We run the risk of breaking this or making a subtle change to the logic in the future which could impact all of KubeVirt.
> 
> Is it possible to get the same results by setting the interface IP from virt-handler instead of virt-controller?
While I'm not sure if we can get Pod info in virt-handler, I don't think if it's a good practice to fetch Pod details from every virt-handler. For a single VM Migrated in the cluster, if the whole cluster's virt-handler hits api/cache for pod details, doesn't seem too promising to me.
Unless we have some way to tell that specific virt-handler to update VMI where it's getting transferred to.

your thoughts? @vladikr @rmohr @fabiand